tests/components/evohome/conftest.py

```diff
@@ -150,6 +161,33 @@ async def setup_evohome(
 
         assert isinstance(mock_client.call_args.kwargs["session"], ClientSession)
 
-        assert mock_client.account_info is not None
+        assert evo and evo.account_info is not None
 
+        mock_client.return_value = evo
         yield mock_client
+
+
+def get_dhw_entity(hass: HomeAssistant, evo: EvohomeClient) -> EvoDHW:
+    """Return the WaterHeater entity of the evohome integration."""
+
+    if (dhw := evo._get_single_tcs().hotwater) is None:
+        pytest.fail("DHW expected, but was not found")
+
+    entity_id = er.async_get(hass).async_get_entity_id(
+        Platform.WATER_HEATER, DOMAIN, dhw._id
+    )
+
+    component: EntityComponent = hass.data.get(Platform.WATER_HEATER)  # type: ignore[assignment]
+    return next(e for e in component.entities if e.entity_id == entity_id)  # type: ignore[return-value]
```

Comment:
We still need to get rid of this one though

Reply:
Oh, I though read-only access would be OK.

How can I get the `entity_id` of the DHW, if all I can get is its `id` and `name`, as in:
```python
        self._attr_unique_id = evo_device.dhwId
        self._attr_name = evo_device.name  # is static
```

Reply:
Oh, I had though reading attrs was OK. I will construct an entity_id instead.